crates/misc/component-async-tests/http/src/lib.rs

```diff
@@ -190,25 +196,20 @@ impl<T: WasiHttpView> wasi::http::types::HostBody for T {
     async fn finish<U>(
         accessor: &mut Accessor<U, Self>,
         this: Resource<Body>,
-    ) -> wasmtime::Result<Result<Option<Resource<Fields>>, ErrorCode>> {
-        let trailers = accessor.with(|mut access| {
-            let trailers = access.table().delete(this)?.trailers;
-            trailers
-                .map(|v| v.read(access.as_context_mut()).map(|v| v.into_future()))
-                .transpose()
+    ) -> wasmtime::Result<FutureReader<Resource<Fields>>> {
+        let trailers = accessor.with(|mut store| {
+            let trailers = store.borrow_mut().table().delete(this)?.trailers;
```

Comment:
I think you could simplify this due to the `DerefMut` implementation.

```suggestion
            let trailers = store.table().delete(this)?.trailers;
```